.github/workflows/lint.yml

```diff
@@ -36,11 +36,13 @@ jobs:
           go-version: 1.19
```

Comment:
@Eileen-Yu,

I was looking this one and I think we have a better way to do that.

See that in our CI we use the lint https://github.com/kubernetes-sigs/kubebuilder/blob/master/.github/workflows/lint.yml#L40 for the Kubebuilder code base.

The github action is nice because is visual and create comments in the pr code to let us know what is wrong.
Then, I checked and it has an option where we can pass the directories 

See:

```
# Optional: working directory, useful for monorepos
          # working-directory: somedir
``` 

From : https://github.com/golangci/golangci-lint-action

So, I think we could:
a) Copy and pate the .github/workflows/lint.yml to .github/workflows/lint_testdata.yml
b) We probably will need to create many jobs one for each testdata project
c) We could have the make lint-testdata and make linte-fix-testdata which will be targets that will do the same that we do for Kubebuilder but for the testdatda projects with a shell script as you did . So that as a dev, we can run them locally to test things out.   

WDYT?

Reply:
Sounds good 👍

I tried to apply reusable workflow to drop duplications. 

I set the reference as the master branch, which supposed to be able to work when this PR gets merged. Would this be good?